README.md

```diff
@@ -32,403 +32,9 @@ sha256sum -c sha256sum.txt 2>&1 | grep OK
-```
-
-If you require a different format than that, you can configure it with the `-request-logging-format` flag.
-The default format is configured as follows:
-
-```
-{{.Client}} - {{.Username}} [{{.Timestamp}}] {{.Host}} {{.RequestMethod}} {{.Upstream}} {{.RequestURI}} {{.Protocol}} {{.UserAgent}} {{.StatusCode}} {{.ResponseSize}} {{.RequestDuration}}
-```
-
-[See `logMessageData` in `logging_handler.go`](./logging_handler.go) for all available variables.
-
-## Adding a new Provider
-
-Follow the examples in the [`providers` package](providers/) to define a new
-`Provider` instance. Add a new `case` to
-[`providers.New()`](providers/providers.go) to allow `oauth2_proxy` to use the
-new `Provider`.
-
-## <a name="nginx-auth-request"></a>Configuring for use with the Nginx `auth_request` directive
-
-The [Nginx `auth_request` directive](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html) allows Nginx to authenticate requests via the oauth2_proxy's `/auth` endpoint, which only returns a 202 Accepted response or a 401 Unauthorized response without proxying the request through. For example:
-
-```nginx
-server {
-  listen 443 ssl;
-  server_name ...;
-  include ssl/ssl.conf;
-
-  location /oauth2/ {
-    proxy_pass       http://127.0.0.1:4180;
-    proxy_set_header Host                    $host;
-    proxy_set_header X-Real-IP               $remote_addr;
-    proxy_set_header X-Scheme                $scheme;
-    proxy_set_header X-Auth-Request-Redirect $request_uri;
-  }
-  location = /oauth2/auth {
-    proxy_pass       http://127.0.0.1:4180;
-    proxy_set_header Host             $host;
-    proxy_set_header X-Real-IP        $remote_addr;
-    proxy_set_header X-Scheme         $scheme;
-    # nginx auth_request includes headers but not body
-    proxy_set_header Content-Length   "";
-    proxy_pass_request_body           off;
-  }
-
-  location / {
-    auth_request /oauth2/auth;
-    error_page 401 = /oauth2/sign_in;
-
-    # pass information via X-User and X-Email headers to backend,
-    # requires running with --set-xauthrequest flag
-    auth_request_set $user   $upstream_http_x_auth_request_user;
-    auth_request_set $email  $upstream_http_x_auth_request_email;
-    proxy_set_header X-User  $user;
-    proxy_set_header X-Email $email;
-
-    # if you enabled --cookie-refresh, this is needed for it to work with auth_request
-    auth_request_set $auth_cookie $upstream_http_set_cookie;
-    add_header Set-Cookie $auth_cookie;
-
-    proxy_pass http://backend/;
-    # or "root /path/to/site;" or "fastcgi_pass ..." etc
-  }
-}
-```
+2.  [Select a Provider and Register an OAuth Application with a Provider](docs/2_auth.md)
+3.  [Configure OAuth2 Proxy using config file, command line options, or environment variables](docs/3_configuration.md)
+4.  [Configure SSL or Deploy behind a SSL endpoint](docs/4_tls.md) (example provided for Nginx)
```

Comment:
I think we should probably add a section called `Documentation` which links to the docs site once it is hosted on GitHub pages